HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -1903,169 +1905,296 @@ def get_poststates(self):
         """
-        Evaluate and report on the Finite Human Wealth Condition
+        Evaluate and report on the Finite Human Wealth Condition.
         """
-
-        self.FHWF = self.PermGroFac[0] / self.Rfree
-        self.cNrmPDV = 1.0 / (1.0 - self.thorn / self.Rfree)
-
         name = "FHWC"
-
-        def test(agent):
-            return self.FHWF < 1
+        FHWFac = self.bilt['FHWFac']
+        result = FHWFac < 1.
 
         messages = {
-            True: "The Finite Human wealth factor value for the supplied parameter values satisfies the Finite Human Wealth Condition.",
-            False: "The given type violates the Finite Human Wealth Condition; the Finite Human wealth factor value is {0.FHWF}",
+            True: f"FHWFac={FHWFac:.5f} : The Finite Human Wealth Factor satisfies the Finite Human Wealth Condition (FHWC) G/R < 1.",
+            False: f"FHWFac={FHWFac:.5f} : The Finite Human Wealth Factor violates the Finite Human Wealth Condition (FHWC) G/R < 1."
         }
+        verbose = self.verbose if verbose is None else verbose
+        self.log_condition_result(name, result, messages[result], verbose)
+        
+    def check_FVAC(self, verbose=None):
+        """
+        Evaluate and report on the Finite Value of Autarky Condition under perfect foresight.
+        """
+        name = "PFFVAC"
+        PFVAFac = self.bilt['PFVAFac']
+        result = PFVAFac < 1.
 
+        messages = {
+            True: f"PFVAFac={PFVAFac:.5f} : The Finite Value of Autarky Factor satisfies the Finite Value of Autarky Condition βG^(1-ρ) < 1.",
+            False: f"PFVAFac={PFVAFac:.5f} : The Finite Value of Autarky Factor violates the Finite Value of Autarky Condition βG^(1-ρ) < 1."
+        }
         verbose = self.verbose if verbose is None else verbose
-        self.check_condition(name, test, messages, verbose)
+        self.log_condition_result(name, result, messages[result], verbose)
+        
+    def describe_parameters(self):
+        '''
+        Make a string describing this instance's parameter values, including their
+        representation in code and symbolically.
+
+        Returns
+        -------
+        param_desc : str
+            Description of parameters as a unicode string.
+        '''
+        params_to_describe = [
+            #[name, description, symbol, time varying]
+            ['DiscFac', 'intertemporal discount factor', 'β',False],
+            ['Rfree', 'risk free interest factor', 'R',False],
+            ['PermGroFac', 'permanent income growth factor', 'G',True],
+            ['CRRA', 'coefficient of relative risk aversion','ρ',False],
+            ['LivPrb', 'survival probability','ℒ',True],
+            ['APFac', 'absolute patience factor', 'Þ=(βℒR)^(1/ρ)',False]
+        ]
+        
+        param_desc = ''
+        for j in range(len(params_to_describe)):
+            this_entry = params_to_describe[j]
+            if this_entry[3]:
+                val = getattr(self,this_entry[0])[0]
```

Comment:
I believe that we now have an alternative way to get at a model's parameters.

I think we are currently storing the parameters dictionary as a dictionary, so that we can pull params from it, rather than relying on `getattr` to pull them from the object. The latter is the "whiteboard problem", and while this still works, I think we've implicitly deprecated this style.

So... consider `self.parameters[this_entry[0]]`